lib/src/parsers/phone_parser.dart

```diff
@@ -102,15 +104,50 @@ abstract class PhoneParser {
         phoneWithoutExitCode.startsWith(callerNationalPrefix)) {
       return callerMetadata;
     }
-    // if no caller was provided we need to make a best guess given the country code
-    final countryCode =
-        CountryCodeParser.extractCountryCode(phoneWithoutExitCode);
-    final national =
-        CountryCodeParser.removeCountryCode(phoneWithoutExitCode, countryCode);
-    // multiple countries use the same country code
-    final metadatas = MetadataFinder.getMetadatasForCountryCode(countryCode);
-    // if multiple countries share the same country code, patterns on the national number
-    // is used to find the correct country.
-    return MetadataMatcher.getMatchUsingPatterns(national, metadatas);
+
+    String? countryCode;
+    Object? countryCodeException;
+    List<PhoneMetadata>? metadatas;
+
+    try {
+      countryCode = CountryCodeParser.extractCountryCode(phoneWithoutExitCode);

```

Comment:
If the phone number did not have an exit code, I believe it is save to assume that there is no country code.

This would prevent issues in germany where phone numbers can start with their corresponding country code